services/harmony/app/models/job.ts

```diff
@@ -465,40 +457,37 @@ export class Job extends DBRecord implements JobRecord {
   }
 
   /**
-   * Returns and array of all the the jobs that are still in the RUNNING state, but have not
-   * been updated in the given number of minutes
+   * Returns the job matching the given query constraints, or null if no such job exists.
    *
-   * @param transaction - the transaction to use for querying
-   * @param minutes - any jobs still running and not updated in this many minutes will be returned
-   * @param currentPage - the index of the page to show
-   * @param perPage - the number of results per page
-   * @returns a list of Job's still running but not updated in the given number of minutes
+   * @param tx - the transaction to use for querying
+   * @param constraints - field / value pairs that must be matched for a record to be returned
+   * @param includeLinks - if true, load all JobLinks into job.links
+   * @param currentPage - the index of the page of links to show
+   * @param perPage - the number of link results per page
+   * @returns the matching job, or null if none exists, along with pagination information
+   * for the job links
    */
-  static async notUpdatedForMinutes(
-    transaction: Transaction,
-    minutes: number,
+  static async queryForSingleJob(
+    tx: Transaction,
+    constraints: JobQuery = {},
+    includeLinks = false,
+    lock = false,
     currentPage = 0,
-    perPage = 10,
-  ):
-    Promise<{ data: Job[]; pagination: ILengthAwarePagination }> {
-    const pastDate = subMinutes(new Date(), minutes);
-    const items = await transaction('jobs')
-      .select()
-      .where({
-        status: JobStatus.RUNNING,
-      })
-      .where('updatedAt', '<', pastDate)
-      .orderBy('createdAt', 'desc')
-      .paginate({ currentPage, perPage, isLengthAware: true });
-
-    const jobs = items.data.map((j) => new Job(j));
-    for (const job of jobs) {
-      job.links = (await getLinksForJob(transaction, job.jobID)).data;
+    perPage = env.defaultResultPageSize,
+  ): Promise<{ job: Job; pagination: ILengthAwarePagination }> {
+    let query = tx('jobs').select().where(constraints.where);
```

Comment:
You probably want to use `first` instead of `select` here since you only want one row. Then you can remove the `[0]` selector on line 483 - `first` returns a single result, not an array.

Reply:
Done